[coverage] skipping code coverage for 'if constexpr' and 'if consteval' (#78033)
llvm/llvm-project: clang/docs/ReleaseNotes.rst

  invalidation by invalid initializer Expr.
  Fixes (`#30908 <https://github.com/llvm/llvm-project/issues/30908>`_)
- Clang now emits correct source location for code-coverage regions in `if constexpr`

>   and `if consteval` branches. Untaken branch is now skipped.

Reviewer: ```suggestion
  and `if consteval` branches. Untaken branches are now skipped.
```
